include/pytorch/tokenizers/regex.h

```diff
@@ -38,11 +38,24 @@ class IRegex {
 };
 
 /**
- * @brief Creates a regex instance. Tries RE2 first, falls back to std::regex.
+ * @brief Creates a regex instance. If no strong symbol defined, only
+ * uses RE2. This is a weak symbol to allow other regex libraries to be
+ * used.
  *
  * @param pattern The regex pattern to compile.
  * @return A unique pointer to an IRegex-compatible object.
  */
 Result<std::unique_ptr<IRegex>> create_regex(const std::string& pattern);
 
+/**
+ * @brief Creates a fallback regex instance. If no strong symbol defined,
+ * returns Error, otherwise uses PCRE2 and std::regex.
```

Comment:
If this is meant to be extensible to other regex libs, would make sense to me to remove PCRE2 from the comment and rename regex_lookahead.cpp to regex_pcre2.cpp or something like that.